lib/mercury/mercury.rb

```diff
@@ -60,17 +60,36 @@ def publish(source_name, msg, tag: '', headers: {}, &k)
     # The amqp gem caches exchange objects, so it's fine to
     # redeclare the exchange every time we publish.
     with_source(source_name) do |exchange|
-      payload = write(msg)
-      pub_opts = Mercury.publish_opts(tag, headers)
-      if publisher_confirms_enabled
-        expect_publisher_confirm(k)
-        exchange.publish(payload, **pub_opts)
-      else
-        exchange.publish(payload, **pub_opts, &k)
-      end
+      publish_internal(exchange, msg, tag, headers, &k)
+    end
+  end
+
+  ORIGINAL_TAG_HEADER = 'Original-Tag'.freeze
```

Comment:
Just to clarify, if you intend for it to be 'internal', you can communicate that by putting them under the private declaration. 

```ruby
class MyClass
  PUBLIC_CONSTANT_ONE = 'one'
  PUBLIC_CONSTANT_TWO = 'two'
  ...

  private

  PRIVATE_CONSTANT_ONE = 'private_one'
  PRIVATE_CONSTANT_TWO = 'private_two'
  ...
end
```